src/AutorestSwift/View Models/ClientMethodOptionsViewModel.swift

```diff
@@ -37,16 +37,17 @@ struct ClientMethodOptionsViewModel {
     /// The name of the options object (i.e. ListContainersOptions)
     let name: String
 
-    let properties: [PropertyViewModel]
+    // The list of properites for the options object
+    let properties: [ParameterViewModel]
 
-    init(from operation: Operation, with model: CodeModel) {
+    /// The variable of the options object in the generated code
+    let variableName: String
+
+    init(from operation: Operation, with model: CodeModel, properties: [ParameterViewModel]) {
```

Comment:
ViewModel initializers shouldn't take other view models as parameters. They should take the underlying model objects.

Reply:
Fixed.